pkg/utils/authcheck/authcheck_test.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authcheck
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	authchecktesting "github.com/google/knative-gcp/pkg/gclient/authcheck/testing"
+)
+
+func TestAuthenticationCheck(t *testing.T) {
+	testCases := []struct {
+		name       string
+		authType   AuthType
+		statusCode int
+		wantError  bool
+		setEnv     bool
+	}{
+		{
+			name:       "authentication check failed, using empty authType",
+			authType:   "empty",
+			statusCode: http.StatusUnauthorized,
+			wantError:  true,
```

Comment:
Was trying to use specific errors, however, in Prow and local, it returns different errors. In local, it can't write error into termination file (no such file), but in Prow, it can. Tried to create file, error showed no permission to do so. Thus, I just verify whether there is an error or not.